app/src/main/java/org/mozilla/fenix/components/metrics/LeanplumMetricsService.kt

```diff
@@ -106,7 +107,9 @@ class LeanplumMetricsService(
         Parser.parseVariables(application)
 
         leanplumJob = scope.launch {
-
+            withContext(Dispatchers.IO) {
+                preferences = application.getSharedPreferences(PREFERENCE_NAME, MODE_PRIVATE)
```

Comment:
This gets initialized too late – it's needed for `Leanplum.setDeviceId` – and results in a unit test failure.

I wonder if we couldn't just move `Leanplum.setDeviceId` into the IO dispatcher – we'd have to make sure it can be set off the main thread and possibly used concurrently first though.

Reply:
We should also ensure earlier code doesn't need it (e.g. `Parser.parseVariables`)